lib/PIClient/PrivacyIDEA.php

```diff
@@ -246,6 +247,151 @@ public function validateCheckWebAuthn(string $username, string $transactionID, s
 		return null;
 	}
 
+	/**
+	 * Authenticate using a passkey. If successful, the response will contain the username.
+	 *
+	 * @param string $transactionID TransactionID.
+	 * @param string $passkeyResponse The json serialized response from the authenticator. Is the same as a webauthnSignResponse.
+	 * @param string $origin Origin of the passkeyResponse, usually gotten from a browser.
+	 * @param array|null $headers Optional headers for the request.
+	 * @return PIResponse|null PIResponse or null if error.
+	 * @throws PIBadRequestException If an error occurs during the request.
+	 */
+	public function validateCheckPasskey(string $transactionID, string $passkeyResponse, string $origin, ?array $headers = null): ?PIResponse
+	{
+		assert(gettype($transactionID) === 'string');
+		assert(gettype($passkeyResponse) === 'string');
+		assert(gettype($origin) === 'string');
+
+		if (!empty($transactionID) && !empty($passkeyResponse) && !empty($origin)) {
+			$passkeyResponseParams = json_decode($passkeyResponse, true);
+
+			if (!is_array($passkeyResponseParams)) {
+				$this->log('debug', 'Invalid passkey response for validateCheckPasskey. Expected an array.');
+				return null;
+			}
+
+			$params = [
+				'transaction_id' => $transactionID,
```

Comment:
could add a constant for transaction_id aswell, since it is used throughout the code